packages/sales-api-service/src/services/__tests__/concession.service.spec.js

```diff
@@ -0,0 +1,283 @@
+        expect(permission.concessions).toContainEqual(getSeniorConcession())
+      })
+
+      it("doesn't add second senior concession if one's already present", async () => {
+        const seniorConcession = getSeniorConcession()
+        const permission = getMockPermission()
+        permission.concessions = [seniorConcession]
+        await concessionService.addSenior(permission)
+        expect(permission.concessions).toEqual([expect.objectContaining(seniorConcession)])
+      })
+
+      it('add senior to blue badge', async () => {
+        const permission = getMockPermission()
+        const disabledConcession = getDisabledNiConcession()
+        permission.concessions = [disabledConcession]
+        await concessionService.addSenior(permission)
+        expect(permission.concessions).toEqual([disabledConcession, getSeniorConcession()])
+      })
+
+      it('add senior to NI', async () => {
+        const permission = getMockPermission()
+        const disabledConcession = getDisabledBlueBadgeConcession()
+        permission.concessions = [disabledConcession]
+        await concessionService.addSenior(permission)
+        expect(permission.concessions).toEqual([disabledConcession, getSeniorConcession()])
+      })
+    })
+
+    describe('removeSenior', () => {
+      it.each([
+        ['has only a senior concession', [getSeniorConcession()], []],
+        [
+          'has senior and disabled concessions',
+          [getSeniorConcession(), getDisabledBlueBadgeConcession()],
+          [getDisabledBlueBadgeConcession()]
+        ]
+      ])('removes senior concession if permission %s', async (_desc, concessions, expectedConcessions) => {
+        const permission = getMockPermission()
+        permission.concessions = concessions
+        await concessionService.removeSenior(permission)
+        expect(permission.concessions).toEqual(expectedConcessions)
+      })
+
+      it.each([
+        ['is empty', []],
+        ['has junior concession', [getJuniorConcession()]],
+        ['has junior and disabled concession', [getJuniorConcession(), getDisabledBlueBadgeConcession()]],
+        ['has disabled concession', [getDisabledNiConcession()]]
+      ])('leaves concessions unmodified if senior concession not present and concessions %s', async (_desc, concessions) => {
+        const permission = getMockPermission()
+        permission.concessions = concessions
+        await concessionService.removeSenior(permission)
+        expect(permission.concessions).toEqual(concessions)
+      })
+    })
+
+    describe('hasSenior', () => {
+      it('is true if licensee has senior concession', async () => {
+        const permission = getMockPermission()
+        permission.concessions = [getSeniorConcession()]
+        const isSenior = await concessionService.hasSenior(permission)
+        expect(isSenior).toBe(true)
+      })
+
+      it("is false if licensee doesn't have senior concession", async () => {
```

Comment:
Maybe overkill, but as we are only testing that it is false if there are no concessions, can we also test that it is false if there is only a non-senior concession? Same goes for tests on the other `has` functions.